src/testdir/test_usercommands.vim

```diff
@@ -304,3 +304,143 @@ func Test_addr_all()
 
   delcommand DoSomething
 endfunc
+
+func Test_command_list()
+  command! DoCmd :
+  call assert_equal("\n    Name              Args Address Complete   Definition"
+        \        .. "\n    DoCmd             0                       :",
+        \           execute('command DoCmd'))
+
+  " Test with various -range= and -count= argument values.
+  command! -range DoCmd :
+  call assert_equal("\n    Name              Args Address Complete   Definition"
+        \        .. "\n    DoCmd             0    .                  :",
+        \           execute('command DoCmd'))
+  command! -range=% DoCmd :
+  call assert_equal("\n    Name              Args Address Complete   Definition"
+        \        .. "\n    DoCmd             0    %                  :",
+        \           execute('command! DoCmd'))
+  command! -range=2 DoCmd :
+  call assert_equal("\n    Name              Args Address Complete   Definition"
+        \        .. "\n    DoCmd             0    2                  :",
+        \           execute('command DoCmd'))
+  command! -count=2 DoCmd :
+  call assert_equal("\n    Name              Args Address Complete   Definition"
+        \        .. "\n    DoCmd             0    2c                 :",
+        \           execute('command DoCmd'))
+
+  " Test with various -addr= argument values.
+  command! -addr=lines DoCmd :
+  call assert_equal("\n    Name              Args Address Complete   Definition"
+        \        .. "\n    DoCmd             0    .                  :",
+        \           execute('command DoCmd'))
+  command! -addr=arguments DoCmd :
+  call assert_equal("\n    Name              Args Address Complete   Definition"
+        \        .. "\n    DoCmd             0    .   arguments      :",
+        \           execute('command DoCmd'))
```

Comment:
The above output and some others below have strange formatting after vim-8.1.1201:  the "Complete" column is not well aligned.  I updated the test to reflect the current vim and I did not try to change formatting in `ex_docmd.c`. But we may consider changing formatting of ":command" again to better align columns.